### ISSUE TITLE
Facebook sharing causes android crash: Attempt to invoke virtual method 'int org.json.JSONArray.length()' on a null object reference

### ISSUE THREAD

**Issue by tommulkins** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#1095]
The error encountered when testing a production build(curious, not emulator) of an apk targeting API 28 returns the following exception when attempting to shareViaFacebook:

```
FATAL EXCEPTION: pool-2-thread-1
    Process: us.worthy.worthy, PID: 8401
    java.lang.NullPointerException: Attempt to invoke virtual method 'int org.json.JSONArray.length()' on a null object reference
        at nl.xservices.plugins.SocialSharing$2.run(SocialSharing.java:271)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
```

**Comment by EddyVerbruggen** [comment]
Not sure why that's an issue all of the sudden, but please try passing in an empty array to the 'files' property (it's probably 'null' now).

**Comment by tommulkins** [comment]
Thanks for that Eddy.  Using the example from the  readme I was using
`window.plugins.socialsharing.shareViaFacebook('My message', null, 'https://my.url')`

Changed it to
`window.plugins.socialsharing.shareViaFacebook('My message', [], 'https://my.url')`

Still the same problem.  

```
java.lang.NullPointerException: Attempt to invoke virtual method 'int org.json.JSONArray.length()' on a null object reference
        at nl.xservices.plugins.SocialSharing$2.run(SocialSharing.java:271)
```

I think the second argument is meant for images and perhaps `shareViaFacebook` method doesn't have a parameter for `files`?

**Comment by devVenus1202** [comment]
I also got the same issue.

**Comment by tommulkins** [comment]
In case anyone else is looking to solve this for recent APIs of Android I fixed using the following:

Just for Android
```
window.plugins.socialsharing.shareVia('com.facebook.katana', 'My message', null, null, 'http://my.url');
```

It appears any of the other methods of sharing directly to Facebook inject an error atm.

**Comment by spinninghamster** [comment]
I am using Android target level 29 with `window.plugins.socialsharing.shareWithOptions` and I am **NOT** experiencing this problem.

**Comment by bongtawo** [comment]
I am using method ACTION_SHARE_VIA_FACEBOOK_WITH_PASTEMESSAGEHINT with this plugin.
android null object problem line 93 in android src file.
so i chaged 'null' to new JSONArray() like below.

from
`return doSendIntent(callbackContext, args.getString(0), args.getString(1), args.isNull(3) ? args.getJSONArray(2) : null, args.getString(3), "com.facebook.katana", null, false, true, "com.facebook.composer.shareintent");`

to
`return doSendIntent(callbackContext, args.getString(0), args.getString(1), args.isNull(3) ? args.getJSONArray(2) : new JSONArray(), args.getString(3), "com.facebook.katana", null, false, true, "com.facebook.composer.shareintent");`

**Comment by eweap** [comment]
@EddyVerbruggen it seems to be fixed on `master`  ! Thanks !

**Comment by ThangD** [comment]
@EddyVerbruggen when 6.0.2 release, thanks.

**Comment by EddyVerbruggen** [comment]
@ThangD It's on npm now.

**Comment by robvandijk** [comment]
Can this be added for `5.6.8` as well? We are experiencing the same crash but cannot update yet to `6.0.2`.

**Comment by EddyVerbruggen** [comment]
@robvandijk 5.6.8 has been published, but 5.6.9 would be possible if you can create a PR.

**Comment by robvandijk** [comment]
PR just created, thanks for looking into this.
